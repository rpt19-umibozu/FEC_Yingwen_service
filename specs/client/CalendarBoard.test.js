import React from 'react';
import Enzyme from 'enzyme';
import 'babel-polyfill';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
import  CalendarBoard  from '../../client/src/CalendarBoard.jsx';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../client/src/CalendarBoard.jsx', () => 'CalendarBoard');

const mockFn = jest.fn();



describe('CalendarBoard Component', () => {

  it('should be defined', () => {
    expect(CalendarBoard).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <CalendarBoard className='calendarFrame' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call mock function when a day is clicked', () => {
    const wrapper = shallow(<CalendarBoard className='calendarFrame' onDayClick={mockFn}/>);

    wrapper.find('.availableDay').simulate('click')
    // expect(mockFn).toHaveBeenCalled();
    expect(wrapper.find('.availableDay').length).toEqual(1);
  })

})
