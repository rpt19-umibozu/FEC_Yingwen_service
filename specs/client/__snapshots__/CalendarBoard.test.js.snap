// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayComponenet Component should render correctly 1`] = `ShallowWrapper {}`;
